core/src/main/java/org/apache/iceberg/ManifestReader.java

```diff
@@ -52,8 +54,13 @@
 public class ManifestReader<F extends ContentFile<F>>
     extends CloseableGroup implements CloseableIterable<F> {
   static final ImmutableList<String> ALL_COLUMNS = ImmutableList.of("*");
-  static final Set<String> STATS_COLUMNS = Sets.newHashSet(
+
+  // the difference between the two stats set below is to support ContentFile.copyWithoutStats(), which
+  // still keeps record count.
+  private static final Set<String> STATS_COLUMNS = Sets.newHashSet(
```

Comment:
When would the caller not want record count?

Reply:
I think `copyWithoutStats` [doesn't discard record count](https://github.com/apache/iceberg/blob/b1296bcbe8e050d4bc28e3d41feb2f8868c8f0bf/core/src/main/java/org/apache/iceberg/BaseFile.java#L167) will discarding all column-specific stats. 

I do agree that having one list is simpler, the reason for me to do this is 
- If we add `record_count` to this list then it will result in a behavior change, that if people select `record_count` without other stats listed here, earlier they will not receive those stats, but now they will receive a full list. This is because `dropStats` relies on this list. 
- Alternatively we can stop copying `recordCount` over within `copyWithoutStats` but I'm not entirely sure if we want to do that since currently the metrics that can be discarded are all map, and `recordCount` is `long`; and I guess if we no longer copy `recordCount ` we may as well not copy `fileSizeInBytes` which is another `long`. After this change since these two attributes return primitive type, they will return -1, which I'm not sure if it's the best thing to do. 

I think the first approach is safer, but I wasn't sure if it's worth changing the behavior to keep the code simpler. Do you have a recommendation?

Reply:
Can we special-case `record_count`? I don't think that `record_count` should be dropped in `copyWithoutStats`, but I also agree that simply selecting `record_count` should not select all stats columns.

This set is primarily for ensuring that all stats required to filter the instances of `DataFile` or `DeleteFile` stored in a manifest are present. I think we should limit it to that purpose. I don't think that this list should affect the behavior or `copyWithoutStats`.

Reply:
Sounds good, I'll add `record_count` to the set so that the set tracks all stats required for filtering data/delete file, and special-case `record_count` in `dropStats()` to account for the case that selecting only `record_count` will still result in `copyWithoutStats` instead of a full copy. (This is actually the same as the current logic, just consolidate the two sets into one)